src/common/Compiler.h

```diff
@@ -95,7 +95,7 @@ int CountTrailingZeroes(unsigned long long x);
 #endif
 
 // GCC and Clang
-#if defined(__GNUC__)
+#if defined(DAEMON_USE_COMPILER_INTRINSICS) && defined(__GNUC__)
```

Comment:
I tried compiling with GCC with an empty definition of NORETURN and as predicted, compilation fails due to errors about functions not returning a value.

This build option is hardly useful if the code can't compile, and IMO disabling the `__attribute__` type of stuff has scant testing value anyway. The scope of `DAEMON_USE_COMPILER_INTRINSICS` would be better restricted to just the BitScanForward type of stuff (that is, if you insist on maintaining a distinction between two types of intrinsics; merging the two under one option would also be good).

Reply:
Weird, on my end it works with both GCC and Clang.

I want a way to test the “unknown compiler” case.

Reply:
The code now always define `NORETURN` and other things if any previous condition failed to define them.